### PR TITLE
Update readme for core kit

### PR DESCRIPTION
**Pull request by jukiginanjar**

### CODE REVIEW

File: MidtransCoreKit/code_usage.md [hunk 1]
```diff
@@ -1,54 +1,296 @@
+
+# Overview
+We provide an API only implementation for all payment types, This allows users to Bring your own UI to the mobile App.
```
Comment (atherlangga) [inline MidtransCoreKit/code_usage.md]: 1. Change `API only` to `API-only`
2. Change `Bring` to `bring`

File: MidtransCoreKit/code_usage.md [hunk 2]
```diff
@@ -1,54 +1,296 @@
...
+
+# Prerequsites
+
+1. Please familiarize yourself with our documentation
```
Comment (atherlangga) [inline MidtransCoreKit/code_usage.md]: The user should not be required to familiarise with VT backend documentation.

File: MidtransCoreKit/code_usage.md [hunk 3]
```diff
@@ -1,54 +1,296 @@
...
 
+It's representation of customer's Order
+
+```
+VTTransactionDetails *transactionDetails =
+[[VTTransactionDetails alloc] initWithOrderID:<#(NSString *)#>
+                               andGrossAmount:<#(NSNumber *)#>];
+```
+
+**Important Note:** `grossAmount` need to be the same as `itemDetails` total price
```
Comment (atherlangga) [inline MidtransCoreKit/code_usage.md]: Add `.., otherwise Veritrans server will reject the transaction.`

File: MidtransCoreKit/code_usage.md [hunk 4]
```diff
@@ -1,54 +1,296 @@
...
+
+
+After you've the data above ready, you can continue to make a payment transaction.
```
Comment (atherlangga) [inline MidtransCoreKit/code_usage.md]: Change to `After the above data ready`

File: MidtransCoreKit/code_usage.md [hunk 5]
```diff
@@ -1,54 +1,296 @@
...
+
+## 2. Normal Credit Card Transaction
+
+In Credit Card transaction there're two main step to make transaction
```
Comment (atherlangga) [inline MidtransCoreKit/code_usage.md]: Change to `there are two main steps`.

File: MidtransCoreKit/code_usage.md [hunk 6]
```diff
@@ -1,54 +1,296 @@
...
+[[VTClient sharedClient] generateToken:tokenRequest completion:^(NSString *token, NSString *redirectURL, NSError *error) {
+    if (error) {
+    	//error generating token
+    } else {
+    	//generate token success
+    }
 }];
+                                       
+```
+**Note:** If you enable [3D Secure](https://en.wikipedia.org/wiki/3-D_Secure) feature, your `Token` are still not valid yet, you need to open `redirectURL` on `VT3DSController`, do the 3D secure things, and if success your `Token` will be valid.
```
Comment (atherlangga) [inline MidtransCoreKit/code_usage.md]: Change to `your token is still in invalid state`, ..

File: MidtransCoreKit/code_usage.md [hunk 7]
```diff
@@ -1,54 +1,296 @@
...
-// PERFORMING TWO CLICK PAYMENT
+## 3. Oneclick Credit Card Transaction
 
-//first generate token id using savedTokenId from VTMaskedCreditCard
-VTTokenRequest *tokenRequest = [VTTokenRequest tokenForTwoClickTransactionWithToken:<savedTokenId from VTMaskedCreditCard>
-                                                                                    cvv:<cvv number>
-                                                                                 secure:<activate 3D secure transaction or not>
-                                                                            grossAmount:<transaction amount>];    
-[[VTClient sharedClient] generateToken:tokenRequest completion:^(NSString *token, NSError *error) {
+One click transaction offer simple credit card transaction because it only need `token` to charge transaction. The `token` represents the credit card number dan expiry date of the customer.
```
Comment (atherlangga) [inline MidtransCoreKit/code_usage.md]: Change to `of the customer's credit card`.

File: MidtransCoreKit/code_usage.md [hunk 8]
```diff
@@ -1,54 +1,296 @@
...
+VTPaymentCreditCard *paymentDetail =
+[[VTPaymentCreditCard alloc] initWithFeature:VTCreditCardPaymentFeatureNormal
+                                       token:token];
+paymentDetail.saveToken = YES;
 
+//next charge transaction process..
+```
+
+#### B. Get Token With Register Credit Card 
+Register credit card provide more simple way to get the oneclick token
```
Comment (atherlangga) [inline MidtransCoreKit/code_usage.md]: Change `more simple` to `simpler`

**Comment by atherlangga** [comment]
What if we change `copy and paste` to `Use following code as a template`?

**Comment by jukiginanjar** [comment]
cc @atherlangga

**Comment by jukiginanjar** [comment]
cc @atherlangga

**Pull request merged**